### DISCUSSION TITLE
Transaction attachment file limit

### DISCUSSION THREAD

**Discussion by ghost** [firefly-iii/firefly-iii#7368] [Firefly III]
I discovered that the max. size in a transaction for an attachment is 1MB. How can I increase the limit?

**Comment by JC5** [comment]
That depends on how you run Firefly III, because the built-in limit is 1GB.

**Comment by ghost** [comment]
Thanks for your reply.

What installation method can I use to bypass this 1MB limit?

Via different sources on the internet I understand that the limit is in Laravel PHP framework and I have to edit a php.ini file for bypassing the limit?

I run Firefly III through Docker.
Do I have to edit something within the Docker container and what do I have to edit?


________________________________
Van: James Cole ***@***.***>
Verstuurd: zaterdag 15 april 2023 06:03
Aan: firefly-iii/firefly-iii ***@***.***>
CC: Danny Janmaat ***@***.***>; Author ***@***.***>
Onderwerp: Re: [firefly-iii/firefly-iii] Transaction attachment file limit (Discussion #7368)


That depends on how you run Firefly III, because the built-in limit is 1GB.

**Reply by JC5** [reply]
I can upload files larger than 1 MB using Docker, the default for Docker is 64mb i think?

**Comment by ghost** [comment]
Hello James,

And are the files really attached in the transaction?

I upload a 2MB image file and it says it attached the file but it's not there. If the file is <1MB it's there.

________________________________
Van: James Cole ***@***.***>
Verstuurd: zaterdag 15 april 2023 09:44
Aan: firefly-iii/firefly-iii ***@***.***>
CC: Danny Janmaat ***@***.***>; Author ***@***.***>
Onderwerp: Re: [firefly-iii/firefly-iii] Transaction attachment file limit (Discussion #7368)


I can upload files larger than 1 MB using Docker, the default for Docker is 64mb i think?

**Reply by JC5** [reply]
Mmm, you might be onto something there haha. Let me validate.

**Reply by JC5** [reply]
Mmm, my local test install has no trouble with this. Let me check it out.

**Comment by ghost** [comment]
Hey James,

Thanks.


________________________________
Van: James Cole ***@***.***>
Verstuurd: zaterdag 15 april 2023 10:25
Aan: firefly-iii/firefly-iii ***@***.***>
CC: Danny Janmaat ***@***.***>; Author ***@***.***>
Onderwerp: Re: [firefly-iii/firefly-iii] Transaction attachment file limit (Discussion #7368)


Mmm, my local test install has no trouble with this. Let me check it out.

**Comment by ghost** [comment]
I found the solution.

I use NGINX to connect to FireFly III. NGINX has a default limit of 1MB.

![image](https://github.com/firefly-iii/firefly-iii/assets/10903083/7b897fbf-e53d-4338-beb6-eed10dc60e7c)

Source: https://www.bookstackapp.com/docs/admin/upload-config/
